src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -787,7 +786,40 @@ public void SaveWallet()
         }
 
         /// <inheritdoc />
-        public bool RemoveTransientTransactions(uint256 depositId = null)
+        public bool RemoveUnconfirmedTransactionData()
+        {
+            lock (this.lockObject)
+            {
+                bool walletUpdated = false;
+
+                foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions.ToList())
+                {
+                    // Change for unconfirmed transaction?
+                    if (transactionData.BlockHeight == null)
+                    {
+                        this.Wallet.MultiSigAddress.Transactions.Remove(transactionData);
+                    }
+                    // Spend by unconfirmed transaction?
+                    else if (transactionData.SpendingDetails != null && transactionData.SpendingDetails.BlockHeight == null)
+                    {
+                        transactionData.SpendingDetails = null;
+                    }
+                    else
+                    {
+                        continue;
+                    }
+
+                    walletUpdated = true;
```

Comment:
this will always be true if there are trx in multisig should this condition not be set in side one of the `if`s

Reply:
The continue in the `else` avoids it being set if its not one of our `if` cases.